File: Java/Hardware.java
// Class Hardware, treated as midclass.
public class Hardware extends Product
{
    // Hardware attributes.
    private String brand;
    private String model;

    /* Constructor. */

    // Empty constructor.
    public Hardware()
    {

    }

    // Constructor with hardware's data (will it be used?).
    public Hardware(String brand, String model)
    {
        this.brand = brand;
        this.model = model;
    }

    /* Setter and Getter. */

    // Set brand attribute.
    public void setBrand(String brand)
    {
        this.brand = brand;
    }

    // Set model attribute.
    public void setModel(String model)
    {
        this.model = model;
    }

    // Get brand attribute.
    public String getBrand()
    {
        return this.brand;
    }

    // Get model attribute.
    public String getModel()
    {
        return this.model;
    }
}

File: C++/hardware.cpp
#include "header.hh"

/* Constructor. */

// Empty constructor.
Hardware::Hardware()
{

}

// Constructor with hardware's data (will it be used?).
Hardware::Hardware(string brand, string model)
{
    this->brand = brand;
    this->model = model;
}

/* Setter and Getter. */

// Set brand attribute.
void Hardware::setBrand(string brand)
{
    this->brand = brand;
}

// Set model attribute.
void Hardware::setModel(string model)
{
    this->model = model;
}

// Get brand attribute.
string Hardware::getBrand()
{
    return brand;
}

// Get model attribute.
string Hardware::getModel()
{
    return model;
}

/* Destructor. */

Hardware::~Hardware()
{
    
}